src/commands/epgsql_cmd_connect.erl

```diff
@@ -119,15 +119,16 @@ hide_password(PasswordFun) when is_function(PasswordFun, 0) ->
     PasswordFun.
 
 
-maybe_ssl(S, false, _, PgSock) ->
+maybe_ssl(S, false, _, PgSock, _Deadline) ->
     epgsql_sock:set_net_socket(gen_tcp, S, PgSock);
-maybe_ssl(S, Flag, Opts, PgSock) ->
+maybe_ssl(S, Flag, Opts, PgSock, Deadline) ->
     ok = gen_tcp:send(S, <<8:?int32, 80877103:?int32>>),
-    Timeout = maps:get(timeout, Opts, 5000),
-    {ok, <<Code>>} = gen_tcp:recv(S, 1, Timeout),
+    Timeout0 = timeout(Deadline),
+    {ok, <<Code>>} = gen_tcp:recv(S, 1, Timeout0),
```

Comment:
would be cool to move this inside a case and handle error in case like
```erlang
case gen_tcp:recv(...) of
  {ok, $S} ->
    …;
  {ok, $N} ->
    …;
  {error, …} ->
    …
end
```

I know `recv` errors were not handled before, but now the probability of getting timeout from it becomes higher.

Reply:
Will fix this!